src/components/Api.js

```diff
@@ -0,0 +1,96 @@
+//Autherization code: 196bd4a5-0b68-4944-a5da-dae91687bd99
+export default class Api {
+  constructor(options) {
+    // constructor body
+    this._baseUrl = options.baseUrl;
+    this._headers = options.headers;
+  }
+
+  //user api
+
+  updateUserInfo(name, about) {
+    return fetch(`${this._baseUrl}/users/me`, {
+      method: "PATCH",
+      headers: this._headers,
+      body: JSON.stringify({
+        name: name,
+        about: about,
+      }),
+    }).then(this._checkResponse);
+  }
+
+  updateProfileAvatar(avatarUrl) {
+    return fetch(`${this._baseUrl}/users/me/avatar`, {
+      method: "PATCH",
+      headers: this._headers,
+      body: JSON.stringify({
+        avatar: avatarUrl,
+      }),
+    }).then(this._checkResponse);
+  }
+
+  getUserInfo() {
+    return fetch(`${this._baseUrl}/users/me`, {
+      headers: this._headers,
+    })
+      .then(this._checkResponse)
+      .catch((err) => {
+        console.error(err);
```

Comment:
You should **not** use `catch` in `Api` because you need to `catch` possible errors at the very **end** of any request, and this end is in `index.js`  after all `then` blocks. And in real projects the `catch` blocks usually open error popups and so on.

Please, remove it from `Api` and add it in `index.js`.